### PR TITLE
Bug 1791739: Restart watches in DC e2e tests

### PR DESCRIPTION
**Pull request by tnozicka**
This makes DC e2e more resilient to apiserver hickups or badly configured loadbalancer.

/cc @soltysh

### CODE REVIEW

File: test/extended/deployments/deployments.go [hunk 1]
```diff
@@ -824,13 +825,14 @@ var _ = g.Describe("[sig-apps][Feature:DeploymentConfig] deploymentconfigs", fun
...
-					}
-					return false, nil
-				})
+			ctx, cancel := context.WithTimeout(context.Background(), deploymentChangeTimeout)
+			defer cancel()
+			_, err = waitForDCModification(ctx, oc.AppsClient().AppsV1(), dc.Namespace, dcName, dc.GetResourceVersion(), func(config *appsv1.DeploymentConfig) (bool, error) {
+				if config.Status.ObservedGeneration >= dc.Generation {
+					return true, nil
+				}
+				return false, nil
```
Comment (soltysh) [inline test/extended/deployments/deployments.go:834]: Nit, this whole block can be simplified to:
```
return config.Status.ObservedGeneration >= dc.Generation, nil
```

**Comment by tnozicka** [comment]
infra flakes
/retest

**Comment by tnozicka** [comment]
yum flake
/retest

**Comment by tnozicka** [comment]
infra
/retest

**Review by soltysh — APPROVED** [review]
/lgtm
/retest

**Pull request merged**